new_course/1_programming_fundementals/basic_enumerable_methods.md

```diff
@@ -1,51 +1,343 @@
-# Basic Enumerable Methods
-* what enumerable methods are
-* some can be used on hashes, others turn a hash into an array
+## Introduction
+
+The Enumerable module is a set of methods that is included in certain Ruby classes (like Array and Hash). You can even include it in your own classes, which will be covered later on. These methods are handy for looping through iterable objects in a collection (most commonly arrays, hashes, and ranges). They provide
+easy ways to perform common actions, and are among the most important tools for Rubyists. Some enumerable methods can be used on hashes just as on arrays, while others are implemented differently depending on the type of collection used.
+
+There is a lot here, but these are built-in to make your life easier. We will run you through the ones that you will most commonly use. This certainly is not an exhaustive list. At the end of this lesson there is a link for you to find out more about other methods beyond what we go through here.
+
+It will be beneficial for you to code along to test the examples as you work through this lesson (in IRB).
```

Comment:
What do you think about adding an additional link here to repl.it as an alternative to IRB? just to give more options to ensure the student is more likely to follow along.